### DISCUSSION TITLE
Release of 16.0.0

### DISCUSSION THREAD

**Discussion by friedpa** [jomjol/AI-on-the-edge-device#3246] [General] [answered]
Dear Developers, do we have a plan for the final release of 16.0.0? When is the SW freeze scheduled, so that we users can start with the release testing?

**Comment by jomjol** [comment]
There is basically one open Label requested for 16.0.0 (basically only a corection of spelling). From my point of view, we could go for 16.0.0 
At the moment I'm testing a new tflite, but this looks good so far.

**Comment by friedpa** [comment]
@jomjol Good Morning, so when can we start testing the RC? Additionally, can we add the Wifi Feature? With the working encryption from @SybexX there should be no reason for being unable to use it... Do you agree?

**Comment by jomjol** [comment]
@friedpa @SybexX : let's start with the RC. Let's not upgrade the tflite. The newest one (710) is worse, than the 710

**Comment by friedpa** [comment]
Today some code change was still added to the rolling... so please just set a date, when we freeze the rolling so that testing can start (and all have the same SW loaded).

**Comment by jomjol** [comment]
@SybexX, @caco3 : I will not add more code. When are you done with current work, so that we freeze for 16.0.0?

**Reply by caco3** [reply]
I was waiting for feedback, but it never came, eg. https://github.com/jomjol/AI-on-the-edge-device/pull/3213
Nevertheless I merged it now.

**Reply by caco3** [reply]
I created now a release candidate: https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v16-0-0-RC1
Please note the open issues: https://github.com/jomjol/AI-on-the-edge-device/issues?q=is%3Aissue+is%3Aopen+label%3ARelease16.0.0

**Reply by caco3** [reply]
Please test and report issues.
If possible, first go to v15.7.0 to check that the migration works for you.

**Comment by friedpa** [comment]
@caco3 Does the red cross mean something?

![grafik](https://github.com/user-attachments/assets/c8ad26c5-71d6-47c9-ae90-3298ddee0eea)

**Reply by caco3** [reply]
Das kannst Du ignorieren.
Da es noch nicht das finale Release ist, soll der Web Installer nicht aktualisiert wrden.
Das siehst Du gut, wenn Du den job aufmachst: https://github.com/jomjol/AI-on-the-edge-device/actions/runs/11096046041
![grafik](https://github.com/user-attachments/assets/fdfaf337-9ff5-4ccf-b25d-51bb76fe6cf6)
Schöner wärs, wenn wir den Job gar nicht ausführen würden, müsste mich da mal genauer damit befassen.

**Reply by caco3** [reply]
NB. This should now be fixed.

**Comment by joeh1242** [comment]
I migrated from 15.7.0. There was much clear picture on 16.0.0 without any setting change:
![image](https://github.com/user-attachments/assets/f9b2fbdc-2184-4696-8350-a898a0bc6ada)

when I opened setting, i have this message:
![image](https://github.com/user-attachments/assets/c00ef6f5-318e-4e51-8bff-7ecae7f3fa4d)

It is not possible to change this parameter.

**Reply by caco3** [reply]
Which RC version did you try?
There was an issue. We are now on RC4 and it should be fixed with that version.

**Reply by joeh1242** [reply]
I had first version. Now it is correct in RC4.  

Will you not add Data file editor #3244 as standard ?

**Reply by caco3** [reply]
@SybexX would have to do this.

**Reply by caco3** [reply]
but if you mess up the file, the graph will no longer work. there is no error handling and no backup!

**Comment by caco3** [comment]
https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v16.0.0-RC5 is out now 🥳

**Reply by Piti61** [reply]
I updated three systems:
- 2x from v16.0.0-RC4: no trouble
- 1x from v15.7.0 (Commit: 0d0b018+): not trouble
Remark: it took two rounds before "Date/Time on device: ..." shows up

**Reply by SybexX** [reply]
@Piti61 Did you press "Ctrl + F5 - Clear the page cache" after the update?

**Reply by Piti61** [reply]
@SybexX Well, I use Safari on my Mac. I pressed the reload button (right to the url) after the update several times, without clearing the cache. Then I checked, if there is a switch in settings/configuration to activate this function. Then I reloaded the overview and "Date/Time" was there. I should mention that I observed this only with the device with v15.7.0 prior to the update.

**Reply by SybexX** [reply]
@Piti61 If the new html files are in the cache, there is no need to delete the cache again.
However, if you on one device e.g. v15.7.0 and v16.0.0 on the others, you always have to do it when you switch between devices.

Google Chrome
- Windows users: hold down Ctrl and then press F5 on your keyboard
- Mac users: hold down Cmd and Shift and then press R on your keyboard

Firefox
- For Windows users: hold down Ctrl and then press F5 on your keyboard
- Mac users: hold down Cmd and Shift and then press R on your keyboard

Safari (Mac)
- Go to Safari > Empty Cache, or hit Opt + Cmd + E
- To refresh, click the refresh button on the address bar or press Cmd + R

**Reply by eugeniumegherea** [reply]
Hey, not sure where to write this feedback, apologies if its in the wrong thread.
I was using for a few months latest stable version 15.7.0 and it had some issues for me, mainly overheating (temp reported around 75C), gui was unusable (the chunks failed to download and requests are usually just stuck in pending), in HA I saw that it didn't report every 5 minutes but every 25-30 minutes, not sure why it skipped.

Yesterday I did install v16 RC5 manually and so far it works way better, the temperatures are sitting around 63C, the image quality is much better, HA integration is now reporting instantly, gui is extremely responsive as well, even when its busy doing the digit detection (I assume high cpu usage should block the gui and web server handling, but it just works). Don't know if it makes sense, but, the full e2e flow from taking picture to HA sending takes aprox 45-60 seconds, again, big improvement. I wonder if it was just thermal throttling constantly n previous release?

I will monitor it further but I must say this is very big improvement

**Reply by caco3** [reply]
> mainly overheating (temp reported around 75C)

The temperature reading can be ignored, it is not reliable at all, specially in the revision 3 boards. Search in the discussions for it.

> gui was unusable (the chunks failed to download and requests are usually just stuck in pending

What is your RSSI?
I guess your wifi is not good enough.
You could have a look on https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-rssithreshold

> it didn't report every 5 minutes but every 25-30 minutes, not sure why it skipped.

Might again be due to weak wifi. We would need a debug log top look into this.

In general it is best to create a new discussion for such questions.

**Reply by eugeniumegherea** [reply]
Thanks, I will create another discussion in the future. For now to provide closure to this, will just reply:
- might get the terminology wrong, but in configuration it is set to a threshold of 75. I didn't change this as far as I remember 
- the wifi signal reported from logs was around 56 last time I checked. It's across the wall and direct distance is approx 3 meters. I looked directly at the logs from SD card because couldn't access them from API. My router is 2.4 and 5 GHz on same ssid name, don't think it's an issue since it worked for a few weeks without issues. It started suddenly.
- about temperature, I've checked the threads and knew about thermals being reported wrong, but anyway it is plausible because the board was getting crazy hot, especially in the area of camera and with this new version it seems to be better

**Comment by Ingo-Knito** [comment]
To the temperature: Some CAMs do not have a (good) thermal paste on the back. Or there was not good thermal connection. This leads to high temperatures.

I’ve bought a thermal tape and put a little piece on the back of the cam to the aluminum shield of the sd card. Now both sides are glued together. This can reduce the temperature of the system significantly.

**Comment by caco3** [comment]
### "Good things are worth waiting for"
In the last months we didn't do a very good job in finalizing the `v16` release.
There where several release candidates, each improving the firmware more and more. Also a lot of new features got added and enhanced. Looking back, using release candidates for this without a proper release plan might not have been the right way to go.
We try to do better next time :smile: 

But now we released [RC7](https://github.com/jomjol/AI-on-the-edge-device/releases/tag/16.0.0-RC7) and agreed on a feature freeze until the final `v16` comes out. So if no new regressions arise, we will change the RC7 to the new release in some weeks.
**Please test it and report any regression.**

**Reply by caco3** [reply, marked as answer]
The final v16.0.0 is out.
🥳
